src/geometry.rs

```diff
@@ -1,9 +1,114 @@
 //! Extension's to Smithay's geometry module.
 
-use std::cmp;
+use std::str::FromStr;
+use std::{cmp, error, fmt, ops};
 
 use smithay::utils::{Point, Size};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Matrix3x3<T: Copy> {
+    storage: Vec<T>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct FromVecError(usize, usize);
+
+impl fmt::Display for FromVecError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Mismatched size when createing object from vector, expected {} got {}",
+            self.0, self.1
+        )
+    }
+}
+
+impl error::Error for FromVecError {}
```

Comment:
Should fully import `Error` and use that everywhere. It's the only `Error` used and it's used all over the place.